packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -1461,14 +1484,72 @@ export class GarbageCollector implements IGarbageCollector {
 				);
 			} else {
 				nodeStateTracker.updateTracking(currentReferenceTimestampMs);
-				if (this.tombstoneMode && nodeStateTracker.state === UnreferencedState.SweepReady) {
-					const nodeType = this.runtime.getNodeType(nodeId);
-					if (nodeType === GCNodeType.DataStore || nodeType === GCNodeType.Blob) {
-						this.tombstones.push(nodeId);
-					}
+				if (nodeStateTracker.state === UnreferencedState.SweepReady) {
+					sweepReadyNodes.push(nodeId);
 				}
 			}
 		}
+
+		return sweepReadyNodes;
+	}
+
+	/**
+	 * Deletes nodes from both the runtime and garbage collection
+	 * @param sweepReadyNodes - nodes that are ready to be deleted
+	 */
+	private runSweepPhase(sweepReadyNodes: string[], gcData: IGarbageCollectionData) {
+		// TODO: GC:Validation - validate that removed routes are not double deleted
+		// TODO: GC:Validation - validate that the child routes of removed routes are deleted as well
+		const sweptRoutes = this.runtime.deleteUnusedNodes(sweepReadyNodes);
+		const updatedGCData = this.deleteSweptRoutes(sweptRoutes, gcData);
+		this.deleteSweptNodeTrackers(sweptRoutes);
+		this.trackDeletedNodes(sweptRoutes);
+		return updatedGCData;
+	}
+
+	/**
+	 * Removes nodes that are swept from GC state and updates the deleted nodes
+	 */
+	private deleteSweptNodeTrackers(gcNodes: string[]) {
```

Comment:
I would combine this and the next two functions into one or inline them into `runSweepPhase`. They all iterate over the swept routes and could be combined.

Reply:
deleteSweptRoutes is different logic from the others, so I'll keep it as a separate function :)